pulp_smash/tests/rpm/api_v2/test_remove_unit.py

```diff
@@ -0,0 +1,283 @@
+        # Package category and package group will differ so we count only RPMs
+        self.assertEqual(
+            set(unit['unit_id'] for unit in bodies[0]
+                if unit['unit_type_id'] == 'rpm'),  # This test is fragile
+            set(unit['unit_id'] for unit in bodies[1]
+                if unit['unit_type_id'] == 'rpm'),  # due to hard-coded
+        )  # indices. But the data is complex, and this makes things simpler.
+
+    def test_unit_removed(self):
+        """Test that correct unit from first repository has been removed."""
+        body = self.responses['units after removal'][0].json()
+        units_names = set(unit['metadata']['name'] for unit in body
+                          if unit['unit_type_id'] == 'rpm')
+        self.assertNotIn(self.removed_unit, units_names)
+
+
+class _RemoveMissingTestCase(_BaseTestCase):
+    """Parent class for RemoveMissingTrueTestCase and RemoveMissingFalseTestCase.
+
+    Provides common functionality shared by both child classes.
+    Following steps are executed:
+        1. Create repository foo with feed, sync and publish it.
+        2. Create repository bar with foo as a feed and run sync.
+        3. Get content of both repositories.
+        4. Remove random unit from repository foo and publish foo.
+        5. Sync repository bar.
+        6. Get content of both repositories.
+
+    """
+
+    @classmethod
+    def setUpClass(cls, remove_missing=False):  # noqa pylint:disable=arguments-differ,line-too-long
+        """Create two repositories, first is feed of second one."""
+        super(_RemoveMissingTestCase, cls).setUpClass()
+        cls.responses = {}
+        cls.task_bodies = {}
+        client = api.Client(cls.cfg, api.safe_handler)
+
+        bodies = tuple((_gen_repo() for _ in range(2)))
+        bodies[0]['importer_config']['feed'] = _FEED_URL
+        repos = []
+        repos.append(client.post(REPOSITORY_PATH, bodies[0]).json())
+        sync_path = urljoin(repos[0]['_href'], 'actions/sync/')
+        # Run sync and wait for the task to complete
+        cls.responses['sync'] = []
+        cls.responses['sync'].append(client.post(
+            sync_path, {'override_config': {}}
+        ))
+        cls.task_bodies['sync'] = []
+        cls.task_bodies['sync'] += tuple(utils.poll_spawned_tasks(
+            cls.cfg, cls.responses['sync'][-1].json()))
+        # Add distributor and publish
+        cls.responses['distribute'] = client.post(
+            urljoin(repos[0]['_href'], 'distributors/'),
+            _gen_distributor(),
+        )
+        cls.responses['publish'] = []
+        cls.responses['publish'].append(client.post(
+            urljoin(repos[0]['_href'], 'actions/publish/'),
+            {'id': cls.responses['distribute'].json()['id']},
+        ))
+
+        # Use http feed instead of https to avoid possible config problems
+        bodies[1]['importer_config']['feed'] = urljoin(
+            # Create http url from base_url
```

Comment:
> I see advantage in testing on content published over http as: If I want to use pulp only over insecure connection, there's no point in using https

I agree that it's nonsensical to make content available over HTTPS and then ignore SSL verification errors. But I've heard from several people who try to use Pulp Smash and immediately get confused when tests fail due to incorrect HTTPS configurations. My solution to that is to let them disable HTTPS verification errors via the configuration file. 

> And I'm not 100% sure about this, but I think that certificate that is configured in pulp smash config file, is used only for talking to pulp's api, while for using secure feed one must correctly configure pulp itself.

Pulp Smash's configuration file cannot be used to configure SSL certificates. It only supports the "verify" option, which controls whether `pulp_smash.api.Client` verifies SSL certificates when requests are made. This class is used for all HTTP and HTTP requests. It's used when making API calls, and it's used when fetching files published by Pulp.

> If I want to use https feed, there should be other tests that verify that everything works.

It's valuable to test that both HTTP and HTTPS publishing works. So let's make a separate test case for that, rather than ignoring the "base_url" set by the user.